lightning/src/util/sweep.rs

```diff
@@ -659,42 +637,76 @@ where
 	}
 }
 
-/// Reads previously persisted spendable output information from the store.
-pub fn read_spendable_outputs<K: Deref, L: Deref>(
-	kv_store: K, logger: L,
-) -> Result<Vec<TrackedSpendableOutput>, io::Error>
+#[derive(Debug, Clone)]
+struct SweeperState {
+	outputs: Vec<TrackedSpendableOutput>,
+	best_block: BestBlock,
+}
+
+impl_writeable_tlv_based!(SweeperState, {
+	(0, outputs, required),
+	(2, best_block, required),
+});
+
+impl<B: Deref, ES: Deref, F: Deref, K: Deref, L: Deref>
+	ReadableArgs<(
+		B,
+		ES,
+		Option<F>,
+		K,
+		L,
+		Box<
+			dyn Fn(&[&SpendableOutputDescriptor]) -> Result<Transaction, ()>
+				+ Send
+				+ Sync
+				+ 'static,
+		>,
+		u32,
+	)> for OutputSweeper<B, ES, F, K, L>
```

Comment:
Instead of using `ReadableArgs` with all these type parameters, should we instead have a `Readable` for the state and just pass that to the `OutputSweeper` constructor? Seems that would remove a lot of boilerplate here and elsewhere.